framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -825,6 +892,16 @@ public Boolean visitTypevar_Typevar(
       } else if (!subtypeHasAnno && !supertypeHasAnno && areEqualInHierarchy(subtype, supertype)) {
         // two unannotated uses of the same type parameter are of the same type
         return true;
+      } else if (subtypeHasAnno) {
```

Comment:
Nit: the ordering of tests in this cascaded if is:

sub && super
!sub && !super && X
sub && !super
super && !sub
!sub && !super && Y

It's a bit weird to see the two cases that include "!sub && !super" separated rather than appearing together (probably at the end).

Also, it might be helpful for documentation (and won't hurt performance) to make the omitted parts of the tests explicit, as I did above.